server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -172,7 +183,14 @@ BytesReference getOrCompute(
      */
     void invalidate(CacheEntity cacheEntity, DirectoryReader reader, BytesReference cacheKey) {
         assert reader.getReaderCacheHelper() != null;
-        cache.invalidate(new Key(cacheEntity, reader.getReaderCacheHelper().getKey(), cacheKey));
+        String readerCacheKeyUniqueId = null;
+        if (reader instanceof OpenSearchDirectoryReader) {
+            IndexReader.CacheHelper cacheHelper = ((OpenSearchDirectoryReader) reader).getDelegatingCacheHelper();
+            readerCacheKeyUniqueId = ((OpenSearchDirectoryReader.DelegatingCacheHelper) cacheHelper).getDelegatingCacheKey()
+                .getId()
+                .toString();
```

Comment:
Given we are mapping UUID to string everywhere, why not store it as such in DelegatingCacheHelper/Key?

Reply:
Yeah make sense.